### DISCUSSION TITLE
pyvista plots are not interactive in VS Code or Jupyter Lab

### DISCUSSION THREAD

**Discussion by mahmouddinar** [pyvista/pyvista#3371] [Q&A]
I recently started using pyvista, and I cannot get the plots to be interactive (orbit controls, pan/zoom) when writing the code in jupyter notebooks and running them either in VS Code, or on a browser through Jupyter, or Jupyter Lab.
The plots are drawn, but they remain static, unlike the code snippets on the documentation, like:

https://docs.pyvista.org/user-guide/jupyter/pythreejs.html

I followed the instructions on this page, but had no success (neither threejs backend, nor vtklink worked). 
I'm working in Windows (sorry!). When I run the same code as a py app, the visualization window is interactive, but of course it's a lot more hassle to write and test the code this way instead of writing it in a notebook. Is this a problem because of Windows, or is there something else? I'm stuck in this first step. Thanks very much.

**Comment by banesullivan** [comment]
You probably need to enable a notebook rendering backend and install it's dependencies. Did you have a chance to look at https://docs.pyvista.org/user-guide/jupyter/index.html#usage-with-pyvista ?


`set_jupyter_backend` is going to be the key here along with installing whatever backend dependencies you wish to use

```py
import pyvista as pv
pv.set_jupyter_backend('pythreejs')  
```

**Comment by mahmouddinar** [comment]
Yes. I had read the documentation, and used set_juyter_backend() with pythreejs and ipyvtklink, and installed dependencies.
I found a workaround on a different forum, and the problem was incompatibility of ipywidgets;
I demoted it to version 7.2.2, and the notebook became interactive, though only with ipyvtklink.

**Reply by banesullivan** [reply]
Okay great, glad you were able to resolve this by downgrading ipywidgets

**Comment by Panjete** [comment]
Faced a similar issue while working on Ubuntu 20.04 and VS Code. #5436 Helped. 

Used `pip install pyvista[jupyter]` within the conda environment and things worked.

**Comment by eric2302** [comment]
None of the above solutions worked for me unfortunately. What worked was calling the following before plotting

```python
import os
import pyvista as pv
os.environ["VTK_DEFAULT_OPENGL_WINDOW"] = "vtkOSOpenGLRenderWindow"
pv.set_jupyter_backend('html')
```
